core/src/toga/widgets/textinput.py

```diff
@@ -224,20 +224,10 @@ def validate(self):
             return False
 
     @property
-    def on_return(self):
-        """The handler to invoke when return is pressed.
+    def on_confirm(self):
+        """The handler to invoke when return is pressed."""
+        return self._on_confirm
 
-        Returns:
-            The function ``callable`` that is called when return is pressed.
-        """
-        return self._on_return
-
-    @on_return.setter
-    def on_return(self, handler):
-        """Set the handler to invoke when return is pressed.
-
-        Args:
-            handler (:obj:`callable`): The handler to invoke when return is pressed.
-        """
-        self._on_return = wrapped_handler(self, handler)
-        self._impl.set_on_return(self._on_return)
+    @on_confirm.setter
+    def on_confirm(self, handler):
+        self._on_confirm = wrapped_handler(self, handler)
```

Comment:
To accomodate the "not implemented" case, add something like:
```
if not getattr(self._impl, "IMPLEMENTS_ON_CONFIRM", True):
    self.factory.not_implemented("TextInput.on_confirm()")
```

Then add `IMPLEMENTS_ON_CONFIRM = False` as a class-level property on Cocoa, etc. 

We still need to retain setting the actual handler for the not-implemented case - the handler can be *set*, we just need to flag that it won't be *used*.

Reply:
And for the systems that do have it implemented, do I need to include the IMPLEMENTS_ON_CONFIRM as well or just on the systems that don't have it implemented yet?

Reply:
Okay, makes sense. Don't have much experience using getattr so wasn't fully aware of it's usage in that way.